js/src/hooks/useAutoCreateAdsMCAccounts.js

```diff
@@ -0,0 +1,185 @@
+/**
+ * External dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import useCreateMCAccount from './useCreateMCAccount';
+import useUpsertAdsAccount from '.~/hooks/useUpsertAdsAccount';
+import useExistingGoogleAdsAccounts from '.~/hooks/useExistingGoogleAdsAccounts';
+import useExistingGoogleMCAccounts from '.~/hooks/useExistingGoogleMCAccounts';
+import useGoogleAdsAccount from './useGoogleAdsAccount';
+import useGoogleMCAccount from './useGoogleMCAccount';
+import { GOOGLE_MC_ACCOUNT_STATUS } from '.~/constants';
+
+/**
+ * Custom hook to handle the creation of Google Merchant Center (MC) and Google Ads accounts.
+ *
+ * @typedef {Object} AutoCreateAccountsStatus
+ * @property {boolean} accountsCreated Indicates if both the Google Ads and Google Merchant Center accounts have been successfully created.
+ * @property {boolean} accountCreationChecksResolved Indicates if the account creation checks (for existing accounts) have been resolved.
+ * @property {boolean} isCreatingAccounts Indicates if the Google Ads and/or Google Merchant Center account(s) are being created.
+ * @property {boolean} isCreatingOnlyAdsAccount Indicates if only the Google Ads account is currently being created.
+ * @property {boolean} isCreatingBothAccounts Indicates if both the Google Ads and Google Merchant Center accounts are currently being created.
+ * @property {boolean} isCreatingOnlyMCAccount Indicates if only the Google Merchant Center account is currently being created.
+ *
+ * @return {AutoCreateAccountsStatus} Object containing properties related to the account creation status.
+ */
+const useAutoCreateAdsMCAccounts = () => {
+	// Refs are used to avoid the re-render of the parent component.
+	const isCreatingBothAccountsRef = useRef( false );
```

Comment:
@ankitrox I don't think there's a need to use refs for:

1. `isCreatingBothAccountsRef`
2. `isCreatingAdsAccountRef`
3. `accountsCreatedRef`

We can use state variables for them which I think is more appropriate since technically we want re renders when those values change. When using refs, there won't be any re renders.